port/linux/omrvmem.c

```diff
@@ -1076,52 +1099,80 @@ get_hugepages_info(struct OMRPortLibrary *portLibrary, vmem_hugepage_info_t *pag
 	return 1;
 }
 
+static BOOLEAN
+set_flags_for_mmap(int *flags)
+{
+	BOOLEAN useBackingFile = FALSE;
+#if defined(MAP_ANONYMOUS)
+	*flags |= MAP_ANONYMOUS | MAP_PRIVATE;
+#elif defined(MAP_ANON)
+	*flags |= MAP_ANON | MAP_PRIVATE;
+#else
+	useBackingFile = TRUE;
+	*flags |= MAP_PRIVATE;
+#endif
+	return useBackingFile;
+}
+
 static void *
-default_pageSize_reserve_memory(struct OMRPortLibrary *portLibrary, void *address, uintptr_t byteAmount, struct J9PortVmemIdentifier *identifier, uintptr_t mode, uintptr_t pageSize, OMRMemCategory *category)
+reserve_memory_with_mmap(struct OMRPortLibrary *portLibrary, void *address, uintptr_t byteAmount, struct J9PortVmemIdentifier *identifier, uintptr_t mode, uintptr_t pageSize, OMRMemCategory *category)
 {
 	/* This function is cloned in J9SourceUnixJ9VMem (omrvmem_reserve_memory).
 	 * Any changes made here may need to be reflected in that version.
 	 */
 	int fd = -1;
-	int ft = -1;
 	int flags = 0;
 	void *result = NULL;
 	int protectionFlags = PROT_NONE;
-	BOOLEAN useBackingSharedFile = FALSE;
 	BOOLEAN useBackingFile = FALSE;
+	BOOLEAN useBackingSharedFile = FALSE;
+	BOOLEAN isHugePagesEnabled = FALSE;
 
-	Trc_PRT_vmem_default_reserve_entry(address, byteAmount);
+	Trc_PRT_vmem_reserve_entry(address, byteAmount);
 
 	if(mode & OMRPORT_VMEM_MEMORY_MODE_SHARE_FILE_OPEN) {
 		flags |= MAP_SHARED;
 		useBackingSharedFile = TRUE;
 	} else {
-#if defined(MAP_ANONYMOUS)
-		flags |= MAP_ANONYMOUS | MAP_PRIVATE;
-		useBackingSharedFile = FALSE;
-#elif defined(MAP_ANON)
-		flags |= MAP_ANON | MAP_PRIVATE;
-		useBackingSharedFile = FALSE;
-#else
-		useBackingFile = TRUE;
-		flags |= MAP_PRIVATE;
-#endif
+		useBackingFile = set_flags_for_mmap(&flags);
 	}
 
+#if defined(MAP_HUGETLB)
+	if(0 != (mode & OMRPORT_VMEM_MEMORY_MODE_HUGE_PAGES)) {
```

Comment:
spacing... 'if (0'. 
there are a couple of other spots in this method that should be fixed.